transport/http/dial.go

```diff
@@ -266,6 +267,14 @@ func defaultBaseTransport(ctx context.Context, clientCertSource cert.Source, dia
 	if trans == nil {
 		trans = fallbackBaseTransport()
 	}
+	trans.DialContext = func(ctx context.Context, network string, addr string) (net.Conn, error) {
+		// Don't try IPv6 if it's not supported.
+		// https://github.com/golang/go/issues/25321
+		if !nettest.SupportsIPv6() {
```

Comment:
I'm a bit hesitant to use a package meant for testing in a production context. Is this what the Go team recommends? Is there any other package we can use that is meant for critical path usage?

Reply:
I think this package would be fine to use as it is meant to test the network and not necessary unit tests. This does feel like code that should happen in terraform though maybe and not here. I think the reason for putting it here is that our transport function is not handling some of the mTLS bits that our NewHTTPClient function does. Is that correct @melinath? If so maybe we can move that logic ad NewHTTPClient calls into NewTransport.

Reply:
> Some of the APIs in that package take testing.T which gave me pause.

I had not noticed that at first glance. That does give me more pause.